handle error scenarios in attestation api

AnieeG opened the pull request:
## Motivation


## Solution

AnieeG committed:
handle error scenarios in attestation api

AnieeG committed:
Dynamic price updates in commit job (#417)

mateusz-sekara commented on core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:
```diff
@@ -139,18 +140,20 @@ func (s *TokenDataReader) ReadTokenData(ctx context.Context, msg internal.EVM2EV
-	messageAndAttestation, err = encodeMessageAndAttestation(messageBody, attestationResp.Attestation)
-	if err != nil {
-		return nil, fmt.Errorf("failed to encode messageAndAttestation : %w", err)
+	switch attestationResp.Status {
```
Can we have a test for that?

roman-kashitsyn committed:
chore(ccipdata): don't use generated types in OffRampReader (#416)

## Motivation

Currently, the `OffRampReader` interface uses the `evm_2_evm_offramp.RateLimiterTokenBucket` type, a generated version-dependent type. The interface abstracts the rest of the system from the version-specific components, so using generated types in its methods is inappropriate.

## Solution

This change updates the interface to use the `ccipdata.TokenBucketRateLimit` type (equivalent to `evm_2_evm_offramp.RateLimiterTokenBucket`) instead. Another interface method, `GetTokenPoolsRateLimits`, already used the version-independent type; this change makes the interface more consistent.

RensR committed:
token pools to support lock/unlock (#421)

winder committed:
ccip: Return an error from ParseLogs. (#408)

AnieeG committed:
Parameterize dest gas limit in tests (#423)

AnieeG committed:
include tests

AnieeG committed:
fix spelling

dimkouv commented on core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:
```diff
@@ -44,6 +44,10 @@ const (
 )
 
+var (
+	ErrAttestionAPI = errors.New("unexpected response from attestation API")
```
maybe rename? `ErrUnknownResponse` ?

AnieeG replied:
updated

dimkouv commented on core/services/ocr2/plugins/ccip/tokendata/usdc/usdc.go:
```diff
@@ -139,18 +144,20 @@ func (s *TokenDataReader) ReadTokenData(ctx context.Context, msg internal.EVM2EV
+		}
+		return messageAndAttestation, nil
+	case attestationStatusPending:
+		return nil, tokendata.ErrNotReady
```
can we include [here](https://github.com/smartcontractkit/ccip/blob/ee7e13aac99729bff32c37832ad2647636fe4685/core/services/ocr2/plugins/ccip/ccipexec/ocr2.go#L400) some handling for `tokendata.ErrNotReady` ? I see it's not used anywhere

```
maybe:
if err != nil {
   if errors.Is(err, tokendata.ErrNotReady) {
      msgLggr.Warnw("skipping message token data not ready", err)
   } else {
      msgLggr.Errorw("skipping message error while getting token data", "err", err) 
   }
   continue
}
```

AnieeG replied:
Added this

AnieeG committed:
review comments

AnieeG committed:
Merge branch 'ccip-develop' into CCIP-1268-handle-attestation-api-errors

AnieeG committed:
Merge branch 'ccip-develop' into CCIP-1268-handle-attestation-api-errors
